tree/dataframe/inc/ROOT/RDF/RVariation.hxx

```diff
@@ -61,91 +54,103 @@ bool ResultsSizeEq(const RVec<T> &results, std::size_t expected, std::size_t nCo
+// GetValuePtr)
+template <typename T>
+void AssignResults(ROOT::RVec<T> &resStorage, ROOT::RVec<T> &&tmpResults)
 {
-   return typeid(Value_t);
+   const auto nVariations = resStorage.size(); // we have already checked that tmpResults has the same size
+
+   for (auto i = 0u; i < nVariations; ++i)
+      resStorage[i] = std::move(tmpResults[i]);
 }
 
-template <typename RVec_t, typename Value_t = typename RVec_t::value_type>
-std::enable_if_t<IsRVec<Value_t>::value, const std::type_info &> GetInnerValueType(std::size_t nCols)
+template <typename T>
+void *GetValuePtrHelper(ROOT::RVec<T> &v, std::size_t /*colIdx*/, std::size_t varIdx)
 {
-   if (nCols == 1) // we are varying one column that is an RVec
-      return typeid(Value_t);
-   else // we are varying multiple columns whose type is the inner type of this RVec
-      return typeid(typename Value_t::value_type);
+   return static_cast<void *>(&v[varIdx]);
 }
 
-// This overload is for the case of a single column and ret_type != RVec<RVec<...>>
+/****** Overloads for the case of multiple columns varied simultaneously *******/
+
 template <typename T>
-void ResizeResults(ROOT::RVec<T> &results, std::size_t /*nCols*/, std::size_t nVariations)
+bool ResultsSizeEq(const T &results, std::size_t expected, std::size_t /*nColumns*/, std::false_type /*isSingleColumn*/)
 {
-   results.resize(nVariations);
+   return std::all_of(results.begin(), results.end(),
+                      [expected](const auto &inner) { return inner.size() == expected; });
 }
 
-// This overload is for the case of ret_type == RVec<RVec<...>>
 template <typename T>
-void ResizeResults(ROOT::RVec<ROOT::RVec<T>> &results, std::size_t nCols, std::size_t nVariations)
+std::size_t GetNVariations(const std::vector<RVec<T>> &results)
 {
-   if (nCols == 1) {
-      results.resize(nVariations);
-   } else {
-      results.resize(nCols);
-      for (auto &rvecOverVariations : results) {
-         rvecOverVariations.resize(nVariations);
-      }
-   }
+   assert(!results.empty());
+   return results[0].size();
 }
 
-// Assign into fLastResults[slot] without changing the addresses of its elements (we gave those addresses away in
-// GetValuePtr)
-// This overload is for the case of a single column and ret_type != RVec<RVec<...>>
 template <typename T>
-void AssignResults(ROOT::RVec<T> &resStorage, ROOT::RVec<T> &&tmpResults, std::size_t /*nCols*/)
+void ResizeResults(std::vector<ROOT::RVec<T>> &results, std::size_t nCols, std::size_t nVariations)
 {
-   const auto nVariations = resStorage.size(); // we have already checked that tmpResults has the same size
-
-   for (auto i = 0u; i < nVariations; ++i)
-      resStorage[i] = std::move(tmpResults[i]);
+   results.resize(nCols);
+   for (auto &rvecOverVariations : results)
+      rvecOverVariations.resize(nVariations);
 }
 
-// This overload is for the case of ret_type == RVec<RVec<...>>
```

Comment:
it's still there: https://github.com/root-project/root/pull/11149/files#diff-02547ca131b8e1327a282bf7be8f97b95fe01ad324825dfd342dfc5dadd307f8R68-R69 (only on one of the two overloads to avoid repetition, but it should be enough)